solana/solana-ibc/programs/solana-ibc/src/validation_context.rs

```diff
@@ -43,12 +49,41 @@ impl ibc::ValidationContext for IbcStorage<'_, '_> {
     }
 
     fn host_height(&self) -> Result<ibc::Height> {
+        #[cfg(feature = "witness")]
+        {
+            let clock =
+                anchor_lang::solana_program::sysvar::clock::Clock::get()
+                    .map_err(|e| ibc::ClientError::ClientSpecific {
+                        description: e.to_string(),
+                    })?;
+
+            let slot = clock.slot;
+            let height = ibc::Height::new(1, slot)?;
+            return Ok(height);
+        }
         let height = u64::from(self.borrow().chain.head()?.block_height);
         let height = ibc::Height::new(1, height)?;
         Ok(height)
```

Comment:
In general, prefer `cfg!` macro to conditional compilation:

```suggestion
        let height = if cfg!(feature = "witness") {
            anchor_lang::solana_program::sysvar::clock::Clock::get()
                .map_err(|e| ibc::ClientError::ClientSpecific {
                    description: e.to_string(),
                })?.slot
        } else {
            u64::from(self.borrow().chain.head()?.block_height)
        };
        ibc::Height::new(1, height)
```

It often makes the code easier to read and makes build failure more
obvious.  With conditional compilation, you need to build with all
combination of features to make sure everything compiles correctly.
With `cfg!` macro, everything is always well formed.

Reply:
i see, that makes sense